pkg/controller/infrastructure/actuator_reconcile.go

```diff
@@ -109,299 +63,43 @@ func (a *actuator) Reconcile(ctx context.Context, infrastructure *extensionsv1al
-			return firewallActionDeleteAndRecreate, nil, nil
-		}
-
-		if fw.Allocation != nil && fw.Allocation.Image != nil && fw.Allocation.Image.ID != nil && *fw.Allocation.Image.ID != r.infrastructureConfig.Firewall.Image {
-			want := r.infrastructureConfig.Firewall.Image
-			image, err := r.mclient.Image().FindLatestImage(image.NewFindLatestImageParams().WithID(want).WithContext(ctx), nil)
-			if err != nil {
-				r.logger.Error(err, "firewall latest image not found", "clustertag", r.clusterTag, "projectid", r.infrastructureConfig.ProjectID, "image", want)
-				return firewallActionDoNothing, nil, &reconciler.RequeueAfterError{
-					Cause:        err,
-					RequeueAfter: 30 * time.Second,
-				}
-			}
-
-			if image.Payload != nil && image.Payload.ID != nil && *image.Payload.ID != *fw.Allocation.Image.ID {
-				r.logger.Info("firewall image has changed", "clusterid", r.clusterID, "machineid", r.machineIDInStatus, "current", *fw.Allocation.Image.ID, "new", *image.Payload.ID)
-				return firewallActionDeleteAndRecreate, nil, nil
-			}
-		}
-
-		currentNetworks := sets.NewString()
-		for _, n := range fw.Allocation.Networks {
-			if *n.Networktype == mn.PrivatePrimaryUnshared || *n.Networktype == mn.PrivatePrimaryShared {
-				continue
-			}
-			if *n.Underlay {
-				continue
-			}
-			currentNetworks.Insert(*n.Networkid)
-		}
-		wantNetworks := sets.NewString()
-		for _, n := range r.infrastructureConfig.Firewall.Networks {
-			wantNetworks.Insert(n)
-		}
-		if !currentNetworks.Equal(wantNetworks) {
-			r.logger.Info("firewall networks have changed", "clusterid", r.clusterID, "machineid", r.machineIDInStatus, "current", currentNetworks.List(), "new", wantNetworks.List())
-			return firewallActionDeleteAndRecreate, nil, nil
-		}
-
-		return firewallActionDoNothing, nil, nil
-	default:
-		err := fmt.Errorf("multiple firewalls exist for this cluster, which is currently unsupported. delete these firewalls and try to keep the one with machine id %q", r.machineIDInStatus)
-		r.logger.Error(
-			err,
-			"multiple firewalls exist for this cluster",
-			"clusterID", r.clusterID,
-			"expectedMachineID", r.machineIDInStatus,
-		)
-		return firewallActionDoNothing, nil, err
-	}
-}
-
-func createFirewall(ctx context.Context, r *firewallReconciler) (machineID string, nodeCIDR string, err error) {
-	nodeCIDR, err = ensureNodeNetwork(ctx, r)
-	if err != nil {
-		r.logger.Error(err, "firewalls node network", "nodecidr", nodeCIDR)
-		return "", "", &reconciler.RequeueAfterError{
-			Cause:        err,
-			RequeueAfter: 30 * time.Second,
-		}
 	}
-
-	err = updateProviderStatus(ctx, r.c, r.infrastructure, r.providerStatus, &nodeCIDR)
+	nodeCIDR, err := ensureNodeNetwork(ctx, networkReconciler)
 	if err != nil {
-		r.logger.Error(err, "firewalls update provider status", "nodecidr", nodeCIDR)
-		return "", "", &reconciler.RequeueAfterError{
+		return &reconciler.RequeueAfterError{
```

Comment:
In other cases we removed the `*reconciler.RequeueAfterError` as it was only printed but not executed. Does it work in this case? (and / or in other occurrences)

Reply:
Seems like I kept the `RequeueAfterError` in the `Reconcile` funcs and just returned errors from the functions are called in `Reconcile`. Despite I have never seen a requeue really taking place, when looking at the underlying code of the extension library, the requeue error should work. So I would assume implementation is ok.